### PR TITLE
Fix Issues in https://github.com/Datura-ai/smart-scrape/pull/192

### PR DESCRIPTION
**Pull request by honestflix**
I fixed issues commented in this PR.
I also updated code to use uids that not included in the history, cause in the code it doesn't use correct uids and functions

### CODE REVIEW

File: neurons/validators/scraper_validator.py [hunk 1]
```diff
@@ -251,6 +253,86 @@ async def run_task_and_score(
 
         return async_responses, uids, event, start_time
 
+    async def run_task_and_score_with_uids(
```
Comment (okradze) [inline neurons/validators/scraper_validator.py:256]: You can use same run_task_and_score function and add initial_uids as optional field

File: neurons/validators/validator.py [hunk 2]
```diff
@@ -206,6 +206,38 @@ async def get_uids(
         # uid_list = list(available_uids.keys())
         return uids.to(self.config.neuron.device)
 
+    async def get_uids_from_list(
+        self,
+        uids,
+        strategy=QUERY_MINERS.RANDOM,
+        is_only_allowed_miner=False,
+        specified_uids=None,
+    ):
```
Comment (okradze) [inline neurons/validators/validator.py:215]: change old get_uids instead of adding new duplicated

File: neurons/validators/scraper_validator.py [hunk 3]
```diff
@@ -454,11 +537,50 @@ async def query_and_score(self, strategy=QUERY_MINERS.RANDOM):
...
+                                "%Y-%m-%dT%H:%M:%SZ"
+                            ).replace(tzinfo=pytz.utc)
+                            
+                            # Check if within last 30 minutes
+                            time_diff_minutes = (current_time - synapse_time).total_seconds() / 60
+                            if time_diff_minutes <= 30 and not is_failed:
+                                uids_with_recent_results.append(uid)
+                                recent_organic_results.append(synapse)
+                                recent_organic_rewards.append(rewards)
+                                break
```
Comment (okradze) [inline neurons/validators/scraper_validator.py:574]: maybe move as function for readability

File: neurons/validators/scraper_validator.py (hunk 1 above)
Comment (honestflix) [reply neurons/validators/scraper_validator.py]: I just updated code to use run_task_and_score with initial uids

File: neurons/validators/validator.py (hunk 2 above)
Comment (honestflix) [reply neurons/validators/validator.py]: done

File: neurons/validators/scraper_validator.py (hunk 3 above)
Comment (honestflix) [reply neurons/validators/scraper_validator.py]: I created get_uids_with_recent_organic_results function in ScraperValidator Class

File: neurons/validators/scraper_validator.py [hunk 4]
```diff
@@ -498,8 +551,18 @@ async def query_and_score(self, strategy=QUERY_MINERS.RANDOM):
...
-            await self.score_random_synthetic_query()
+                        
+            await self.compute_rewards_and_penalties(
+                event=event,
+                tasks=tasks,
+                responses=final_synapses,
+                uids=uids,
+                start_time=start_time,
+                is_synthetic=True,
+            )
```
Comment (okradze) [inline neurons/validators/scraper_validator.py:562]: Use `score_random_synthetic_query` as before

**Pull request closed without merging**